internal/provider/data_source_packer_image_iteration_test.go

```diff
@@ -130,50 +137,160 @@ func upsertBuild(t *testing.T, bucketSlug, fingerprint, iterationID string) {
 		CloudProvider: "aws",
 		ComponentType: "amazon-ebs.example",
 		IterationID:   iterationID,
-		Status:        models.HashicorpCloudPackerBuildStatusDONE,
-		Images: []*models.HashicorpCloudPackerImage{
-			{
-				ImageID: "ami-42",
-			},
-			{
-				ImageID: "ami-43",
+		Status:        models.HashicorpCloudPackerBuildStatusRUNNING,
+	}
+
+	build, err := client.Packer.CreateBuild(createBuildParams, nil)
+	if err, ok := err.(*packer_service.CreateBuildDefault); ok {
+		t.Errorf("unexpected CreateBuild error, expected nil or 409. Got %v", err)
+	}
+
+	if build == nil {
+		t.Errorf("unexpected CreateBuild error, expected non nil build response. Got %v", err)
```

Comment:
I had a leftover bucket from my last run called `alpine`, and when I changed the test bucket ID to that name, I got a conflict error. But the test didn't stop at the error and continued on until it panicked. So I think you'll want to add a `return` to all these error cases.
```suggestion
		return t.Errorf("unexpected CreateBuild error, expected non nil build response. Got %v", err)
```

Reply:
Ah good catch. thanks for testing. I made a few updates that should allow you to clean up any left over buckets. 

The changes to update the channel seemed to find a bug with the handling of channels that have no assigned iterations -see [here](https://github.com/hashicorp/terraform-provider-hcp/pull/180/files#diff-149bb019e4ee75954203b371d7496a1343410e41cf1f1292e3d60e915a8e491dR167) 👍